### ISSUE TITLE
alm2cl: lmax_out has no effect

### ISSUE THREAD

**Issue by DanielLenz** [healpy/healpy#474]
`hp.alm2cl` takes the `lmax` (based on the size of the input a_lm) and the  `lmax_out` (lmax of the output Cl) keywords, but doesn't use the `lmax_out` keyword.

From _sphtools.pyx:

```cython

    #######################
    # Computing the spectra
    #
    cdef int j, l, m, limit
    cdef int lmax_ = lmax, mmax_ = mmax
    cdef int lmax_out_ = lmax_out

    cdef np.ndarray[double, ndim=1] powspec_
    cdef np.ndarray[np.complex128_t, ndim=1] alm1_
    cdef np.ndarray[np.complex128_t, ndim=1] alm2_

    spectra = []
    for n in xrange(Nspec): # diagonal rank
        for m in xrange(0, Nspec - n): # position in the diagonal
            powspec_ = np.zeros(lmax + 1)
            alm1_ = alms[m]
            alm2_ = alms2[m + n]
            # compute cross-spectrum alm1[n] x alm2[n+m]
            # and place result in result list
            for l in range(lmax_ + 1):
                j = alm_getidx(lmax_, l, 0)
                powspec_[l] = alm1_[j].real * alm2_[j].real
                limit = l if l <= mmax else mmax
                for m in range(1, limit + 1):
                    j = alm_getidx(lmax_, l, m)
                    powspec_[l] += 2 * (alm1_[j].real * alm2_[j].real +
                                        alm1_[j].imag * alm2_[j].imag)
                powspec_[l] /= (2 * l + 1)
            spectra.append(powspec_)

```

It's an easy fix, but should also be properly included in the tests.

**Comment by DanielLenz** [comment]
I've fixed this and written tests, but cannot install healpy due to the issues in https://github.com/healpy/healpy/issues/455

**Comment by zonca** [comment]
@DanielLenz unfortunately I don't have a mac os machine at hand to test that. Have you tried some of the fixes proposed in #455?

**Comment by DanielLenz** [comment]
I've tried some, not all. I'll go ahead and set up a clean docker environment, that should remove any conflicts.
